src/Tokenizer.js

```diff
@@ -91,20 +91,27 @@ export function slice (begin, end) {
 	return substr(temporary, begin, end)
 }
 
+/**
+ * @return {number}
+ */
+export function resetcursor() {
+	return line = column = 1, position = 0
+}
+
 /**
  * @param {string} value
  * @return {number}
  */
 export function alloc (value) {
-	return line = column = 1, length = strlen(temporary = value), position = 0
+	return length = strlen(temporary = value), resetcursor()
 }
 
 /**
  * @param {object[]} value
  * @return {object[]}
  */
 export function dealloc (value) {
-	return temporary = '', value
+	return temporary = '', resetcursor(), value
```

Comment:
this was needed to produce correct line mappings to the created root node (from what I understand), as it's created **after** `dealloc` call (because it's the one returning root's children) the cursor is already at the end, but we want to have line mappings pointing to the beginning